### PR TITLE
Feature/signal gain

### PR DESCRIPTION
**Pull request by Kmudsam01**
* **Tickets addressed:** 511
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  
* 
## Description
This pull request will implement the ApplySignalGain function, which takes an image path, the signal gain the image was taken with, and the desired signal gain. This function creates a new image with the desired signal gain by reversing the gain the image was taken with, and applying the new gain.

## Verification
Changes were validated by applying the function to a test image and inspecting the resulting image.

## Documentation
No documentation change is needed.

## Future work
Estimation of signal gain, via spectral reflectance estimation.

### CODE REVIEW

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp [hunk 1]
```diff
@@ -22,7 +22,97 @@ void URenderingFunctionsLibrary::ApplyCosmicRays()
...
+	Image = Image * (1 / ImageGain);
+
+	//Init Resulting image
+	cv::Mat ResultImage = cv::Mat(Image.rows, Image.cols, Image.type());
+
+	ResultImage = Image * DesiredGain;
+
+	//Clamp values to [0,255]
+	//Separate color channels
+	std::vector<cv::Mat> DifferentColorChannels;
```
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.cpp]: :small_red_triangle_down: Are the number of color channels fixed? If so we should use `std::array<cv::Mat, 3>` or similar

File: Source/cielim/Private/RenderingFunctionsLibrary.h [hunk 2]
```diff
@@ -22,21 +22,12 @@ class  CIELIM_API URenderingFunctionsLibrary :
...
-
-	
-};
-
-
+	static FString ApplyReadNoise(FString Filepath, float ReadNoiseSigma, float SystemGain);
 
+	UFUNCTION(BlueprintCallable, Category="Rendering Functions")
+	static FString ApplySignalGain(FString Filepath, float ImageGain, float DesiredGain);
+};
```
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.h]: :small_red_triangle_down: New line at end of file

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp (hunk 1 above)
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.cpp]: Done! changed it to std::array<cv::Mat, 3>

File: Source/cielim/Private/RenderingFunctionsLibrary.h (hunk 2 above)
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.h]: New line -> added

**Review by patkenneally — APPROVED** [review]

**Pull request merged**